src/js/utils/css.js

```diff
@@ -101,33 +101,41 @@ define([
         });
     };
 
-    var hexToRgba = function (hexColor, opacity) {
+    var canvasColorContext;
+    var getRgba = function (color, opacity) {
         var style = 'rgb';
-        if (hexColor) {
-            hexColor = String(hexColor).replace('#', '');
-            if (hexColor.length === 3) {
-                hexColor = hexColor[0] + hexColor[0] + hexColor[1] + hexColor[1] + hexColor[2] + hexColor[2];
-            }
-        } else {
-            hexColor = '000000';
-        }
-        var channels = [
-            parseInt(hexColor.substr(0, 2), 16),
-            parseInt(hexColor.substr(2, 2), 16),
-            parseInt(hexColor.substr(4, 2), 16)
-        ];
-        if (opacity !== undefined && opacity !== 100) {
+        var hasAlpha = (opacity !== undefined && opacity !== 100);
+        if (hasAlpha) {
             style += 'a';
-            channels.push(opacity / 100);
         }
-        return style + '(' + channels.join(',') + ')';
+        if (!canvasColorContext) {
```

Comment:
It lets us use any CSS color like "red", not just hex, and it's fast enough that I'm not concerned about performance impact. There may be a little time/memory hit getting the context. After that it's pretty clean - we'd have to run tests to see which is slower and which cause more GC as a result of memory usage.